pypeit/pypeit.py

```diff
@@ -499,21 +512,50 @@ def reduce_exposure(self, frames, bg_frames=None, std_outfile=None):
                           f'that failed was {self.caliBrate.failed_step}.  Continuing by '
                           f'skipping this detector.')
                 continue
-            # Extract
+
+            calibrated_det.append(self.det)
+            calib_slits.append(self.caliBrate.slits)
+            # initial_sky and skymask are needed in the extract loop
+            global_sky, sobjs_obj, skymask = self.reduce_one(frames, self.det, bg_frames, std_outfile=std_outfile,
+                                                             objfind_run=True)
+            if len(sobjs_obj)>0:
```

Comment:
This seems inconsistent to me. You are appending the intermediate products, which I understand, but you are generating the SciImg fresh each time, which I don't agree with. It seems we should be storing those as well, otherwise there will be huge performance issues.